pkg/server/transform.go

```diff
@@ -212,3 +217,72 @@ func translationFromProto(t *translatev1.Translation) (*model.Translation, error
 func translationsToProto(m []model.Translation) []*translatev1.Translation {
 	return sliceToProto(m, translationToProto)
 }
+
+// ----------------------Mask----------------------
+
+func maskFromProto(message proto.Message, mask *fieldmaskpb.FieldMask) (model.Mask, error) {
+	if mask == nil {
+		return nil, nil
+	}
+
+	paths := mask.GetPaths()
+
+	if len(paths) == 0 {
+		return model.Mask{}, nil
```

Comment:
Always return nil slice. Mostly, an empty slice is required only when marshalling JSON.

```suggestion
		return nil, nil
```

Reply:
it changes functionality:` nil mask != empty mask`

nil mask means update dst with all values from src
empty mask means update nothing, e.g. leave dst as is

Reply:
nil, is when in request fieldMask is not specified, eg left out
empty is when it is specified in the request, but left empty

Reply:
looks confusing to me. why would anybody want to update without updating? empty (nil) means implicit full update. nice and clear. what do you think?

Reply:
>  If a FieldMask object is not present in a get operation, the
 operation applies to all fields (as if a FieldMask of all fields
 had been specified).

https://protobuf.dev/reference/csharp/api-docs/class/google/protobuf/well-known-types/field-mask

Same goes for update operations.

in the document, there is no information on how API should behave when an empty fieldmask is received, so I infer that an empty mask is a valid mask, and updates nothing.

> . The API is required to only change the values of the fields as specified in the mask and leave the others untouched.

no paths == leave everything untouched

Of course, we could change the implementation, so if field mask is provided it must contain at least one path, else throw an error, but by doing so we are violating design...

Reply:
Answering, on this
> why would anybody want to update without updating?

why would one provide fieldmask field in request, and leave it empty. You either provide it and fill paths you want to update, or don't provide it at update everything 🤷

Reply:
The intent is the same whether the field mask was set empty or not set at all, isn't it?

Reply:
who knows the one's intent in this scenario :)

Reply:
Simplicity is your sherpa :)

Reply:
I would go with the error option then
if fieldmask not provided -> update all 
if fieldmask provided and empty -> error fieldmask should contain at least one path

It seems the most correct/logical one, at least for me